File: test/fixtures/sample-code/documented.ts
/**
 * Calculates the total price including tax.
 * @param items - Array of item prices
 * @param tax - Tax rate as a decimal
 * @returns Total price with tax
 */
export function calculateTotal(items: number[], tax: number): number {
  const subtotal = items.reduce((sum, item) => sum + item, 0);
  return subtotal * (1 + tax);
}

/**
 * A user profile returned from the API.
 */
export interface UserProfile {
  id: string;
  name: string;
  email: string;
}

File: test/fixtures/sample-code/documented.py
def calculate_total(items: list[float], tax: float) -> float:
    """Calculate the total price including tax.

    Parameters
    ----------
    items : list[float]
        List of item prices.
    tax : float
        Tax rate as a decimal.

    Returns
    -------
    float
        Total price with tax applied.
    """
    subtotal = sum(items)
    return subtotal * (1 + tax)


class ShoppingCart:
    """A shopping cart that tracks items and computes totals."""

    def __init__(self):
        self.items = []
